src/backend_model_instance.cc

```diff
@@ -204,6 +206,17 @@ TritonModelInstance::SetInstances(
       existing_instances = model->IndexInstances();
 
   static triton::common::HostPolicyCmdlineConfig empty_host_policy;
+  std::vector<std::future<Status>> creation_results;
+
+  // Deferred will be lazily evaluated when the result is requested. Since the
+  // creation_results are requested serially below, this is equivalent to making
+  // the calls serially.
+  auto launch_policy = std::launch::deferred;
+  // If the backend supports it, std::launch::async will allow the calls to be
+  // made concurrently.
+  if (model->Backend()->BackendAttributes().parallel_instance_loading_) {
+    launch_policy = std::launch::async;
+  }
```

Comment:
Note, I thought this would be a clean way of handling the if/else behavior of whether the parallel loading is supported or not. However, I understand that the `deferred` logic may not be intuitive to reason as the same as "sequential/serial" loading.

If you don't like this approach, I can simplify it to be more straightforward, maybe something like this.
```cpp
// Used to gather async future results, useless for non-parallel case
std::vector<std::future<Status>> creation_results;
...
// assume some lambda or helper func exists for InstanceCreate logic

for each instance {
    if (parallel_loading_supported) {
      creation_results.emplace_back(std::async(std::launch::async, InstanceCreateHelper(args...)));
    } else {
      RETURN_IF_ERROR(InstanceCreateHelper(args...));
    }
}

if (parallel_loading_supported) {
  for result in creation_results {
    // check results, set retval if any failures
  }
}

return Status::Success;
```